api/webhooks/webhooks.py

```diff
@@ -102,47 +154,116 @@ def _call_webhook(
         headers.update({FLAGSMITH_SIGNATURE_HEADER: signature})
 
     try:
-        return requests.post(
+        res = requests.post(
             str(webhook.url), data=json_data, headers=headers, timeout=10
         )
+        res.raise_for_status()
+        return res
     except requests.exceptions.RequestException as exc:
         logger.debug("Error calling webhook", exc_info=exc)
         raise
 
 
-def _call_webhook_email_on_error(
-    webhook: WebhookModels, data: typing.Mapping, webhook_type: WebhookType
+@register_task_handler()
+def call_webhook_with_failure_mail_after_retries(
+    webhook_id: int,
+    data: typing.Mapping,
+    webhook_type: str,
+    send_failure_mail: bool = False,
+    max_retries: int = settings.WEBHOOK_BACKOFF_RETRIES,
+    try_count: int = 1,
 ):
+    """
+    Call a webhook with support for sending failure emails after retries.
+
+    :param webhook_id: The ID of the webhook to be called.
+    :param data: A mapping containing the data to be sent in the webhook request.
+    :param webhook_type: The type of the webhook to be triggered.
+    :param send_failure_mail: Whether to send a failure notification email (bool, default is False).
+    :param max_retries: The maximum number of retries to attempt (int, default is 3).
+    :param try_count: Stores the current retry attempt count in scheduled tasks,
+                        not needed to be specified (int, default is 1).
+    """
+
+    if try_count > max_retries:
+        raise ValueError("try_count can't be greater than max_retries")
+
+    if webhook_type == WebhookType.ORGANISATION.value:
+        webhook = OrganisationWebhook.objects.get(id=webhook_id)
+    else:
+        webhook = Webhook.objects.get(id=webhook_id)
+
+    headers = {"content-type": "application/json"}
+    json_data = json.dumps(data, sort_keys=True, cls=DjangoJSONEncoder)
+    if webhook.secret:
+        signature = sign_payload(json_data, key=webhook.secret)
+        headers.update({FLAGSMITH_SIGNATURE_HEADER: signature})
+
     try:
-        res = _call_webhook(webhook, data)
-    except requests.exceptions.RequestException as exc:
-        send_failure_email(
-            webhook,
-            data,
-            webhook_type,
-            f"N/A ({exc.__class__.__name__})",
+        res = requests.post(
+            str(webhook.url), data=json_data, headers=headers, timeout=10
         )
+        res.raise_for_status()
+    except requests.exceptions.RequestException as exc:
+        if try_count == max_retries:
+            if send_failure_mail:
+                send_failure_email(
+                    webhook,
+                    data,
+                    webhook_type,
+                    f"N/A ({exc.__class__.__name__})",
```

Comment:
Error display heuristics were lost: 

```suggestion
                    f"{exc.response.status_code if exc.response else 'N/A'} ({exc.__class__.__name__})",
```